python/tskit/trees.py

```diff
@@ -2250,14 +2250,15 @@ def map_mutations(self, genotypes, alleles):
 
         The state reconstruction is returned as two-tuple, ``(ancestral_state,
         mutations)``, where ``ancestral_state`` is the allele assigned to the
-        tree root(s) and ``mutations`` is a list of :class:`Mutation` objects.
+        tree root(s) and ``mutations`` is a list of :class:`Mutation` objects,
+        ordered as :ref:`required in a mutation table<sec_mutation_requirements>`.
         For each mutation, ``node`` is the tree node at the bottom of the branch
         on which the transition occurred, and ``derived_state`` is the new state
-        after this mutation. When multiple mutations are returned the ``parent``
-        property contains the index of the previous mutation on the path to root
-        (see the :ref:`sec_mutation_table_definition` for more information on the
-        concept of mutation parents). All other attributes of the :class:`Mutation`
-        object are undefined and should not be used.
+        after this mutation. The ``parent`` property contains the index in the
+        returned list of the previous mutation on the path to root, or ``tskit.NULL``
```

Comment:
It's not the index in the returned list, it's the parent mutation's ID (i.e., it's index in the mutation table)

Reply:
I don't think it can be, can it (which is why I wanted to clarify this). `map_mutations` has no concept of a pre-existing mutation table, does it? After all, the mutations that it returns have not yet been added to the table, so they don't really have an ID yet?

Reply:
[Here](https://github.com/tskit-dev/tsinfer/blob/ab5cddc9f498f28eb6034b1c16a1178f36aafa4e/tsinfer/inference.py#L637) for instance, I explicitly use `mutation.parent` as an index into the returned list, which needs to be converted into an id in the mutation table.

Reply:
Sorry, my bad, I didn't look at the context. I'll look at this again next week - to much going on right now!

Reply:
No hurry!